ivadomed/scripts/extract_small_dataset.py

```diff
@@ -42,11 +42,11 @@ def is_good_contrast(fname, good_contrast_list):
 
 
 def remove_some_contrasts(folder, subject_list, good_contrast_list):
-    file_list = [os.path.join(folder, s, "anat", f) for s in subject_list
-                 for f in os.listdir(os.path.join(folder, s, "anat"))]
-    rm_file_list = [f for f in file_list if not is_good_contrast(f, good_contrast_list)]
+    file_list = [Path(folder, s, "anat", f) for s in subject_list
```

Comment:
I would add IDE useful type hint here for `file_list: List[Path]`. Don't forget to `from typing import List`. 

I would also recommend break everything within [ ] (including the next row) for readability/debugging etc.  list comprehensive while fast here is going to be very painful to step through and debug.